hydra-node/src/Hydra/Chain/Direct/State.hs

```diff
@@ -834,18 +850,22 @@ genCloseTx numParties = do
   ctx <- genHydraContextFor numParties
   (u0, stOpen) <- genStOpen ctx
   snapshot <- genConfirmedSnapshot 0 u0 (ctxHydraSigningKeys ctx)
-  pointInTime <- genPointInTime
+  let (UnsafeContestationPeriod cp) = ctxContestationPeriod ctx
+      contestationPeriodSlotNo = fromIntegral cp
   cctx <- pickChainContext ctx
-  pure (cctx, stOpen, close cctx stOpen snapshot pointInTime, snapshot)
+  (startSlot, pointInTime) <- genSlotWithPointInTimeFromCP contestationPeriodSlotNo
+  -- XXX: The difference between startSlot and pointInTime needs to be <= contestationPeriod
+  pure (cctx, stOpen, close cctx stOpen snapshot startSlot pointInTime, snapshot)
 
 genContestTx :: Gen (HydraContext, PointInTime, ClosedState, Tx)
 genContestTx = do
   ctx <- genHydraContextFor 3
   (u0, stOpen) <- genStOpen ctx
   confirmed <- genConfirmedSnapshot 0 u0 []
-  closePointInTime <- genPointInTime
   cctx <- pickChainContext ctx
-  let txClose = close cctx stOpen confirmed closePointInTime
+  let (UnsafeContestationPeriod cp) = Hydra.Chain.Direct.State.contestationPeriod cctx
+  (startSlot, closePointInTime) <- genSlotWithPointInTimeFromCP $ fromIntegral cp
```

Comment:
This also interprets contestation period seconds as slots and is wrong for our devnet!